### PR TITLE
feat(object-file): added ability to apply JSON patches to object files

### PR DESCRIPTION
**Pull request by tompahoward**
---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/object-file.test.ts [hunk 1]
```diff
@@ -306,3 +307,83 @@ describe("addToArray", () => {
     });
   });
 });
+
+describe("addJsonPatch", () => {
+  test("addJsonPatch(p, v) adds to an existing array", () => {
```
Comment (Chriscbr) [inline test/object-file.test.ts:312]: I think the test names need to be updated based on the new API names

**Review by Chriscbr — COMMENTED** [review]
I like the idea of shortening the name from `addJsonPatch()` to `patch()` 🙂 

Looks great, just found one nitpick

**Review by mrgrain — CHANGES_REQUESTED** [review]
@tompahoward Looking forward to this if you'll find the time to get it over the line =)

File: test/object-file.test.ts (hunk 1 above)
Comment (tompahoward) [reply test/object-file.test.ts]: done... finally 🎉

**Review by mrgrain — APPROVED** [review]
Looking good now! Thanks for fixing up the final bits 😃 

:shipit:

**Pull request merged**